src/Services/PatientTransactionService.php

```diff
@@ -0,0 +1,292 @@
+        $referToId = $this->getUserIdByNpi($data["referToNpi"]);
+
+        $sql = 
+        "
+            INSERT INTO lbt_data (form_id, field_id, field_value) VALUES
+            (?,  'body', ?),
+            (?,  'refer_date', ?),
+            (?,  'refer_diag', ?),
+            (?,  'refer_from', ?),
+            (?,  'refer_to', ?),
+            (?,  'refer_risk_level', ?),
+            (?,  'refer_vitals', ?),
+            (?,  'refer_authorization', ?),
+            (?,  'refer_visits', ?),
+            (?,  'refer_validFrom', ?),
+            (?,  'refer_validThrough', ?)
+        ";
+
+        $params = array
+        (
+            $transactionId, $data["body"],
+            $transactionId, $data["referralDate"],
+            $transactionId, $data["referDiagnosis"],
+            $transactionId, $referById,
+            $transactionId, $referToId,
+            $transactionId, strtolower($data["riskLevel"]),
+            $transactionId, $data["includeVitals"],
+            $transactionId, $data["authorization"],
+            $transactionId, $data["visits"],
+            $transactionId, $data["validFrom"],
+            $transactionId, $data["validThrough"],
+        );
+
+        $results = sqlInsert(
+            $sql,
+            $params
+        );
+
+        if (!$results) {
+            return false;
+        }
+
+        return $results;
+    }
+
+    public function update($tid, $data){
+        
+        $referById = $this->getUserIdByNpi($data["referByNpi"]);
+        $referToId = $this->getUserIdByNpi($data["referToNpi"]);
+        $body = $data["body"];
+        $referralDate = $data["referralDate"];
+        $referralDiagnosis = $data["referDiagnosis"];
+        $riskLevel = strtolower($data["riskLevel"]);
+        $includeVitals = $data["includeVitals"];
+        $authorization = $data["authorization"];
+        $visits = $data["visits"];
+        $validFrom = $data["validFrom"];
+        $validThrough = $data["validThrough"];
+
+        $this->UpdateTransactionForm($tid, 'refer_from', $referById);
+        $this->UpdateTransactionForm($tid, 'refer_to', $referToId);
+        $this->UpdateTransactionForm($tid, 'body', $body);
+        $this->UpdateTransactionForm($tid, 'refer_date', $referralDate);
+        $this->UpdateTransactionForm($tid, 'refer_diag', $referralDiagnosis);
+        $this->UpdateTransactionForm($tid, 'refer_risk_level', $riskLevel);
+        $this->UpdateTransactionForm($tid, 'refer_vitals', $includeVitals);
+        $this->UpdateTransactionForm($tid, 'refer_authorization', $authorization);
+        $this->UpdateTransactionForm($tid, 'refer_visits', $visits);
+        $this->UpdateTransactionForm($tid, 'refer_validFrom', $validFrom);
+        $this->UpdateTransactionForm($tid, 'refer_validThrough', $validThrough);
+        
+        return $this->getOneFromDb($tid);
+    }
+
+    public function UpdateTransactionForm($formId, $fieldId, $value){
```

Comment:
should be lower case u in function name:
updateTransactionForm

Reply:
resolved